src/AxelarGateway.sol

```diff
@@ -43,6 +49,45 @@ abstract contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
         _;
     }
 
+    /******************\
+    |* Public Methods *|
+    \******************/
+
+    function sendToken(
+        uint256 destinationChain,
+        string memory symbol,
+        uint256 amount
+    ) external returns (bool) {
+        address tokenAddress = tokenAddresses(symbol);
+        require(tokenAddress != address(0), 'TOKEN_NOT_EXIST');
+
+        TokenType tokenType = _getTokenType(symbol);
+        bytes memory tokenCallData;
+        bytes32 salt;
+
+        if (tokenType == TokenType.External) {
+            tokenCallData = abi.encodeWithSelector(IERC20.transferFrom.selector, msg.sender, address(this), amount);
+        } else if (tokenType == TokenType.InternalBurnableFrom) {
+            tokenCallData = abi.encodeWithSelector(BurnableMintableCappedERC20.burnFrom.selector, msg.sender, amount);
+        } else if (tokenType == TokenType.InternalBurnable) {
+            salt = keccak256(abi.encodePacked(destinationChain, msg.sender, tokenAddress));
+            address depositAddress = BurnableMintableCappedERC20(tokenAddress).depositAddress(salt);
+
+            tokenCallData = abi.encodeWithSelector(IERC20.transferFrom.selector, msg.sender, depositAddress, amount);
+        }
+
+        (bool success, bytes memory returnData) = tokenAddress.call(tokenCallData);
+        require(success && (returnData.length == uint256(0) || abi.decode(returnData, (bool))), 'BURN_FAIL');
+
+        if (tokenType == TokenType.InternalBurnable) {
+            BurnableMintableCappedERC20(tokenAddress).burn(salt);
+        }
+
+        emit TokenSent(destinationChain, symbol, amount);
```

Comment:
If the contract cannot validate `destinationChain`, what does axelar network do if it doesn't view it as one of those being supported due to whatever reason, e.g. misconfiguration or user error?

Reply:
That's a very good point. I think we need to add dynamic adding and removing allowed destination chains

Reply:
Added support for that